### DISCUSSION TITLE
Approach on checking if one or more fields are dirty

### DISCUSSION THREAD

**Discussion by undesicimo** [react-hook-form/react-hook-form#11107] [Q&A]
As I checked `isDirty` state only returns `true` when all fields is changed from their initial values.
I fetch data and set them as  default values on my form to have them edited. In this case a partial change on the values is a plausible.
Is the following code the best way to approach checking partial updates on my form?
```
const handleSubmit = (values)=>{
if (Object.keys(form.formState.dirtyFields).length === 0) {
....
  }
```

**Comment by Moshyfawn** [comment]
#1991
